File: EnumGeneration.Domain/Enums.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EnumGeneration.Domain
{
    public enum Jedi
    {
        Youngling = 0,
        Padawan = 1,
        Knight = 2,
        Master = 3
    }

    public enum LightSaber
    {
        Blue = 0,
        Green = 1,
        Red = 2,
        Purple = 3
    }

    public enum TIESeriesFigther
    {
        [Description("TIE Starfighter")]
        TIEStarfighter = 0,
        [Description("TIE/LN Starfighter (Light Duty)")]
        TIELNStarfighter_Light_Duty = 1,
        [Description("Super TIE")]
        SuperTIE = 2,
        [Description("TIE/IN Interceptor (Emperor's Royal Guard)")]
        TIEINInterceptor_Emperors_Royal_Guard = 3,
        [Description("TIE Hunter")]
        TIEHunter = 4,
        [Description("TIE Heavy Bomber")]
        TIEHeavyBomber = 5
    }
}

File: EnumGeneration.WebApi/Scripts/Enums.js
var Jedi = {
    Youngling: function () { var fn = function () { return 0; }; fn.Text = "Youngling"; fn.Value = 0; return fn; }(),
    Padawan: function () { var fn = function () { return 1; }; fn.Text = "Padawan"; fn.Value = 1; return fn; }(),
    Knight: function () { var fn = function () { return 2; }; fn.Text = "Knight"; fn.Value = 2; return fn; }(),
    Master: function () { var fn = function () { return 3; }; fn.Text = "Master"; fn.Value = 3; return fn; }(),
};
var LightSaber = {
    Blue: function () { var fn = function () { return 0; }; fn.Text = "Blue"; fn.Value = 0; return fn; }(),
    Green: function () { var fn = function () { return 1; }; fn.Text = "Green"; fn.Value = 1; return fn; }(),
    Red: function () { var fn = function () { return 2; }; fn.Text = "Red"; fn.Value = 2; return fn; }(),
    Purple: function () { var fn = function () { return 3; }; fn.Text = "Purple"; fn.Value = 3; return fn; }(),
};